### ISSUE TITLE
Uncaught Error: Call to undefined method Adldap\Models\Contact::getGroupNames()

### ISSUE THREAD

**Issue by ianharrier** [munkireport/munkireport-php#1081]
This seems to be similar to [another closed issue](https://github.com/munkireport/munkireport-php/issues/974).

I'm using AD authentication, which works fine in MunkiReport 2.15.2.2795, but in 3.1.1.3370, I get the following error and stack trace:

```
Fatal error: Uncaught Error: Call to undefined method Adldap\Models\Contact::getGroupNames() in /srv/www/app/app/lib/munkireport/AuthAD.php:31

Stack trace:
#0 /srv/www/app/app/lib/munkireport/AuthHandler.php(54): munkireport\lib\AuthAD->login('USERNAME', 'PASSWORD')
#1 /srv/www/app/app/lib/munkireport/AuthHandler.php(34): munkireport\lib\AuthHandler->validate('USERNAME', 'PASSWORD')
#2 /srv/www/app/app/controllers/auth.php(80): munkireport\lib\AuthHandler->login('USERNAME', 'PASSWORD')
#3 /srv/www/app/system/kissmvc_core.php(88): munkireport\controller\Auth->login()
#4 /srv/www/app/system/kissmvc.php(17): KISS_Engine->__construct(Array, 'show', 'index', 'AUTO')
#5 /srv/www/app/public/index.php(147): Engine->__construct(Array, 'show', 'index', 'AUTO')
#6 {main} thrown in /srv/www/app/app/lib/munkireport/AuthAD.php on line 31
```

From my testing, it seems that this issue is only occurring when there is a contact object with a similar name to a user object in our domain. For example:

User object name: **First Last**
Contact object name: **First Last (mobile)**

**Comment by mosen** [comment]
I've pushed a change which now uses userPrincipalName as the filter criteria if account_suffix is empty. It seems like adldap2 still uses objectClass=Person which AD Contacts also have! To omit contacts therefore the objectClass filter would have to be ``user``, because both contact and user have person and organizationalPerson classes.

**Comment by mosen** [comment]
woops! bochoven was right. adldap2 will be fixed upstream.

**Comment by ianharrier** [comment]
@mosen Is the change you pushed included in 3.2.0, and should it be considered an adequate workaround for this issue until Adldap2 is fixed? If so, I'll try it out. Thanks!

**Comment by ianharrier** [comment]
This issue appears to be resolved in MunkiReport 3.2.5.3516. Thanks!
